fixieai/cli/agent/commands.py

```diff
@@ -400,44 +416,27 @@ def deploy(ctx, path, metadata_only):
 
     if config.deployment_url is None and not metadata_only:
         # Deploy the agent to fixie with some bootstrapping code.
-        file_streams: Dict[str, BinaryIO] = {}
-        deploy_root = os.path.dirname(path)
-        for root, dirs, files in os.walk(deploy_root, topdown=True):
-            # Exclude any hidden files by removing them from the dirs list.
-            # As per https://docs.python.org/3/library/os.html#os.walk
-            # removing them from the dirs list with topdown=True skips
-            # them from the walk.
-            for i, dir in reversed(list(enumerate(dirs))):
-                if dir.startswith("."):
-                    console.print(
-                        f"Ignoring hidden directory {os.path.join(root, dir)}",
-                        style="grey53",
-                    )
-                    # N.B. We're iterating in reverse to avoid needing to adjust i.
-                    del dirs[i]
-
-            for filename in files:
-                if filename.startswith("."):
-                    console.print(
-                        f"Ignoring hidden file {os.path.join(root, filename)}",
-                        style="grey53",
-                    )
-                    continue
-
-                full_path = os.path.join(root, filename)
-                upload_path = os.path.join(
-                    "agent", os.path.relpath(full_path, deploy_root)
+        with tempfile.TemporaryFile() as tarball_file:
+            with tarfile.open(fileobj=tarball_file, mode="w:gz") as tarball:
+                deploy_root = os.path.dirname(path)
+                tarball.add(
```

Comment:
does this do the right thing on windows?

Reply:
It does indeed! Good question :)